tests/Model/IndexInfoFunctionalTest.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace MongoDB\Tests\Model;
+
+use MongoDB\Collection;
+use MongoDB\Operation\CreateIndexes;
+use MongoDB\Operation\ListIndexes;
+use MongoDB\Tests\Operation\FunctionalTestCase;
+
+class IndexInfoFunctionalTest extends FunctionalTestCase
+{
+    private $collection;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->collection = new Collection($this->manager, $this->getDatabaseName(), $this->getCollectionName());
+    }
+
+    public function testIs2dSphere()
+    {
+        $this->collection->createIndex(['pos' => '2dsphere']);
+        $result = $this->collection->listIndexes();
+
+        $result->rewind();
+        $this->assertInstanceOf('MongoDB\Model\IndexInfo', $result->current());
```

Comment:
This is asserting the first IndexInfo for the `_id` index, which we then ignore when advancing. Perhaps we can just remove this, as `testListIndexesForNewlyCreatedCollection()` in ListIndexesFunctionalTest already tests IndexInfoIterator in this regard. I'd propose we instead capture the name returned by `createIndex()` and then assert that it's the same as `$index->getName()` below to ensure we're dealing with the expected index record.

Alternatively, we can just remove this and go right to asserting `is2dSphere()` on the second element.